autopilot/graph.go

```diff
@@ -192,30 +194,25 @@ func (d *databaseChannelGraph) addRandChannel(node1, node2 *btcec.PublicKey,
 	}
 
 	var lnNode1, lnNode2 *btcec.PublicKey
-	node1Bytes := vertex1.PubKey.SerializeCompressed()
-	node2Bytes := vertex2.PubKey.SerializeCompressed()
-	if bytes.Compare(node1Bytes, node2Bytes) == -1 {
-		lnNode1 = vertex1.PubKey
-		lnNode2 = vertex2.PubKey
+	if bytes.Compare(vertex1.PubKeyBytes[:], vertex2.PubKeyBytes[:]) == -1 {
+		lnNode1, _ = vertex1.PubKey()
+		lnNode2, _ = vertex2.PubKey()
 	} else {
-		lnNode1 = vertex2.PubKey
-		lnNode2 = vertex1.PubKey
+		lnNode1, _ = vertex2.PubKey()
+		lnNode2, _ = vertex1.PubKey()
 	}
 
 	chanID := randChanID()
 	edge := &channeldb.ChannelEdgeInfo{
-		ChannelID:   chanID.ToUint64(),
-		NodeKey1:    lnNode1,
-		NodeKey2:    lnNode2,
-		BitcoinKey1: vertex1.PubKey,
-		BitcoinKey2: vertex2.PubKey,
-		Capacity:    capacity,
+		ChannelID: chanID.ToUint64(),
+		Capacity:  capacity,
 	}
+	edge.AddNodeKeys(lnNode1, lnNode2, lnNode1, lnNode2)
```

Comment:
I think this changes the behavior of the previous statement. Before, the bitcoin keys are _always_ assigned as vertex1 then vertex2, even if the node keys are flipped. Here the bitcoin keys will take on the same ordering as the node keys.

Reply:
The bitcoin keys aren't of importance as we only look at the node keys. This is for generating random graphs during tests.

Reply:
Ahh i see!